benchmark/util/TransformVector.h

```diff
@@ -0,0 +1,164 @@
+the type of the entries in `SourceVectorType` and return a value of the type
+of the entries in `TargetVectorType`.
+
+@param sourceVector The vector to transform.
+*/
+template<typename SourceVectorType, typename TargetVectorType,
+typename TranslationFunction>
+static TargetVectorType transformVectorImpl(
+  SourceVectorType& sourceVector,
+  TranslationFunction translationFunction){
+  // We just use `transformVectorAndAppendImpl` on an empty vector and return
+  // that.
+  TargetVectorType targetVector{};
+  transformVectorAndAppendImpl(sourceVector, &targetVector,
+    translationFunction);
+  return targetVector;
+}
+
+/*
+Concept, that a function `UnaryFunction` takes one argument of a specified typ
+and returns a value of an other specified type.
+*/
+template<typename UnaryFunction, typename SourceType, typename TargetType>
+concept UnaryFunctionWithSpecificTypes = requires(UnaryFunction func, SourceType s){
+  {func(s)} -> std::same_as<TargetType>;
+};
+
+/*
+@brief Transforms the content of a given vector with a given function and
+appends it to another given vector.
+
+@tparam SourceVectorContentType, TargetVectorContentType The type of the
+objects, that the vectors hold.
+@tparam TranslationFunction Must be a function, that takes one argument of
+type SourceVectorContentType and return a value of TargetVectorContentType.
+
+@param sourceVector The vector to transform.
+@param targetVecctor The vector to append to.
+*/
+template<typename SourceVectorContentType, typename TargetVectorContentType,
+typename TranslationFunction>
+requires UnaryFunctionWithSpecificTypes<TranslationFunction,
+SourceVectorContentType, TargetVectorContentType>
+void transformVectorAndAppend(
+  std::vector<SourceVectorContentType>& sourceVector,
+  std::vector<TargetVectorContentType>* const targetVector,
+  TranslationFunction translationFunction){
+
+  transformVectorAndAppendImpl(sourceVector, targetVector, translationFunction);
+}
+
+/*
+@brief Transforms the content of a given vector with a given function and
+appends it to another given vector.
+
+@tparam SourceVectorContentType, TargetVectorContentType The type of the
+objects, that the vectors hold.
+@tparam TranslationFunction Must be a function, that takes one argument of
+type SourceVectorContentType and return a value of TargetVectorContentType.
+
+@param sourceVector The vector to transform.
+@param targetVecctor The vector to append to.
+*/
+template<typename SourceVectorContentType, typename TargetVectorContentType,
+typename TranslationFunction>
+requires UnaryFunctionWithSpecificTypes<TranslationFunction,
+SourceVectorContentType, TargetVectorContentType>
+void transformVectorAndAppend(
+  const std::vector<SourceVectorContentType>& sourceVector,
+  std::vector<TargetVectorContentType>* const targetVector,
+  TranslationFunction translationFunction){
+
+  transformVectorAndAppendImpl(sourceVector, targetVector, translationFunction);
+}
```

Comment:
I think most of this is a rather verbose duplicate of stuff we already have in `src/util/Algorithm.h`.
We have `transform` and `appendToVector`. If you really need the concatenation (transformAndAppend), then you can add it there.

Reply:
Done, see [e11445e](https://github.com/ad-freiburg/qlever/pull/860/commits/e11445e85f0c3d18e9c2560245acb6de9762e540).